server/src/main/java/org/elasticsearch/indices/recovery/plan/ShardSnapshotsService.java

```diff
@@ -94,9 +93,7 @@ public void fetchLatestSnapshotsForShard(ShardId shardId, ActionListener<Optiona
         logger.debug("Searching for peer recovery compatible snapshots in [{}]", repositories);
 
         GetShardSnapshotRequest request = GetShardSnapshotRequest.latestSnapshotInRepositories(
-            clusterService.state().getMinTransportVersion().onOrAfter(TransportVersions.SNAPSHOT_REQUEST_TIMEOUTS)
-                ? TimeValue.MINUS_ONE
-                : TimeValue.MAX_VALUE,
+            MasterNodeRequest.infiniteMasterNodeTimeout(clusterService.state().getMinTransportVersion()),
```

Comment:
The version constant used is changed here. But it should be OK since we are just going to use max_value for a few more versions before 8.14, right?

Reply:
Yeah I thought the change to support `MINUS_ONE` as an infinite timeout landed in 8.15 but actually it was 8.14 so this is fine.